### PR TITLE
#164798023 Implement follow and unfollow feature

### PR DESCRIPTION
**Pull request by amaechi-chuks**
#### Description
 This PR keep tabs of users following each other and by extension the articles they author.

#### Type of change
- [x] New feature

#### How Has This Been Tested?
- [x] unit testing
- [x] Hit this callback endpoint `http://localhost:6000/api/v1/follow/:followeeId` and received a response `Successfully followed user` from the server
- [x] Hit the callback endpoint `http://localhost:6000/api/v1/follow/unFolloweeId` and received a response `successfully unfollowed a user`
#### Checklist:
- [x] create controller functions(callback and redirect)
- [x] write mock test using `chai and mocha`
- [x] implement validations
#### PT-ID
[164798023](https://www.pivotaltracker.com/story/show/164798023)

#### Screenshots
**follow a user**
image

**unfollow a user**
image

####  Background
N/A

### CODE REVIEW

File: server/controllers/follow.controllers.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        });
+      }
+      // Is follower already following followee?
+      const getllFollowerRecord = await Follower.findAll();
+      const userFollowing = getllFollowerRecord
+        .map(follower => follower)
+        .filter(
+          followed =>
+            followed.follower_id === followerId &&
+            followed.followee_id === followeeId
```
Comment (rotimi-babalola) [inline server/controllers/follow.controllers.js]: Use a more meaningful variable name in the `map` function. Also what is the meaning of the variable `userFollowing`? @amaechi-chuks

**Review by rotimi-babalola — COMMENTED** [review]
In general, let's try to have JSDocs for each function also explaining the difference between `follower` and `followee` it's a bit confusing to understand what's happening

File: server/controllers/follow.controllers.js (hunk 1 above)
Comment (amaechi-chuks) [reply server/controllers/follow.controllers.js]: The variable userFollowing is a validation that checks for follower and followee in the database.

Comment (rotimi-babalola) [reply server/controllers/follow.controllers.js]: > The variable userFollowing is a validation that checks for follower and followee in the database.

Can you describe what the variable is going to contain? Is it an object or array? And what will be its content(s)? @amaechi-chuks

File: server/controllers/follow.controllers.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const followController = {
+  /**
+   * @description user follow and unfollow each other
+   * @param {*} req
+   * @param {*} res
+   * @param {followeeId} user to be followed
+   * @param {followerId} person following the user
+   * @returns {object} response and user profile data
+   */
+  async followUser(req, res) {
```
[the remark this answers is not in the document]
Comment (rotimi-babalola) [reply server/controllers/follow.controllers.js:18]: Have you seen this? @amaechi-chuks

**Pull request merged**